record-builder-core/src/main/java/module-info.java

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright 2019 Jordan Zimmerman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+module io.soabase.recordbuilder.core {
```

Comment:
What about this instead (see: https://github.com/tfesenko/Java-Modules-JPMS-CheatSheet#shortcut-an-open-module)?

```
open module io.soabase.recordbuilder.core {
}
```

Reply:
Module which uses annotations still need to be able to import them for compilation, so `exports` directive is required.

I'm not sure how annotation processors work with JPMS, but it looks like that module providing annotations doesn't need to be open.

I've tested this with Gradle 7.0M3 and this worked with standard configuration, e.g.: 

```
plugins {
    `java-library`
}

repositories {
    mavenLocal()
    mavenCentral()
}

dependencies {
    annotationProcessor("io.soabase.record-builder:record-builder-processor:1.20-SNAPSHOT")
    compileOnly("io.soabase.record-builder:record-builder-core:1.20-SNAPSHOT")
}
```

Reply:
I'll test with Maven later today or tomorrow to validate.